File: final_dataset_A/generated_code/claude/python/order_status_update_endpoint_detailed-claude.py
from flask import Flask, request, jsonify
import datetime

app = Flask(__name__)

orders_db = {
    "1001": {"status": "pending", "last_updated": None},
    "1002": {"status": "shipped", "last_updated": None}
}

VALID_STATUSES = {"pending", "processing", "shipped", "delivered", "cancelled"}


@app.route('/api/orders/<order_id>/status', methods=['PUT'])
def update_order_status(order_id):
    data = request.get_json(silent=True)
    if not data or 'status' not in data:
        return jsonify({"status": "error", "message": "Missing status field"}), 400

    new_status = data['status']

    if order_id not in orders_db:
        return jsonify({"status": "error", "message": "Order not found"}), 404

    if new_status not in VALID_STATUSES:
        return jsonify({"status": "error", "message": "Invalid status value"}), 400

    orders_db[order_id]['status'] = new_status
    orders_db[order_id]['last_updated'] = datetime.datetime.utcnow().isoformat()

    return jsonify({
        "status": "success",
        "order_id": order_id,
        "new_status": new_status,
        "updated_at": orders_db[order_id]['last_updated']
    })


if __name__ == '__main__':
    app.run(debug=True)

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/OrderStatusUpdateEndpointDetailed-claude.java
package com.example.order;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;

@RestController
@RequestMapping("/api/orders")
public class OrderStatusUpdateEndpointDetailed {

    private final Map<String, Map<String, String>> ordersDb = new HashMap<>();
    private static final Set<String> VALID_STATUSES = Set.of("pending", "processing", "shipped", "delivered", "cancelled");

    public OrderStatusUpdateEndpointDetailed() {
        Map<String, String> order1 = new HashMap<>();
        order1.put("status", "pending");
        ordersDb.put("1001", order1);

        Map<String, String> order2 = new HashMap<>();
        order2.put("status", "shipped");
        ordersDb.put("1002", order2);
    }

    @PutMapping("/{orderId}/status")
    public ResponseEntity<Map<String, Object>> updateOrderStatus(
            @PathVariable String orderId, @RequestBody Map<String, String> body) {

        Map<String, Object> response = new HashMap<>();

        if (body == null || !body.containsKey("status")) {
            response.put("status", "error");
            response.put("message", "Missing status field");
            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
        }

        String newStatus = body.get("status");

        if (!ordersDb.containsKey(orderId)) {
            response.put("status", "error");
            response.put("message", "Order not found");
            return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
        }

        if (!VALID_STATUSES.contains(newStatus)) {
            response.put("status", "error");
            response.put("message", "Invalid status value");
            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
        }

        String updatedAt = Instant.now().toString();
        ordersDb.get(orderId).put("status", newStatus);
        ordersDb.get(orderId).put("last_updated", updatedAt);

        response.put("status", "success");
        response.put("order_id", orderId);
        response.put("new_status", newStatus);
        response.put("updated_at", updatedAt);
        return ResponseEntity.ok(response);
    }
}
